types/tablet_v2/wlr_tablet_v2_tool.c

```diff
@@ -0,0 +1,515 @@
+#ifndef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 200809L
+#endif
+
+#include "tablet-unstable-v2-protocol.h"
+#include <assert.h>
+#include <stdlib.h>
+#include <types/wlr_tablet_v2.h>
+#include <wayland-util.h>
+#include <wlr/types/wlr_tablet_tool.h>
+#include <wlr/types/wlr_tablet_v2.h>
+#include <wlr/util/log.h>
+
+static void handle_tablet_tool_v2_set_cursor(struct wl_client *client,
+		struct wl_resource *resource, uint32_t serial,
+		struct wl_resource *surface_resource,
+		int32_t hotspot_x, int32_t hotspot_y) {
+	struct wlr_tablet_tool_client_v2 *tool = tablet_tool_client_from_resource(resource);
+	if (!tool) {
+		return;
+	}
+
+	struct wlr_surface *surface = NULL;
+	if (surface_resource != NULL) {
+		surface = wlr_surface_from_resource(surface_resource);
+		if (wlr_surface_set_role(surface, "wl_pointer-cursor", surface_resource,
```

Comment:
I think the `error_resource` for this call is the client resource, not the surface resource.